### PR TITLE
Upgrade pinecone-ts-client to v1.1.0

### LINKED ISSUE #2437: Upgrade Pinecone TypeScript client
The latest new and improved Pinecone TypeScript version is `0.1.6-dev`.

### PR DESCRIPTION
**Pull request by zackproser**
Fixes #2437

These changes upgrade the Pinecone TypeScript integration to v1.1.0. Our [migration guide](https://github.com/pinecone-io/pinecone-ts-client/blob/main/v1-migration.md) is here. In general, the upgrade to v1.1.0 brings: 
* Improved and simplified patterns for instantiating a Pinecone client
* Better error handling and messages

### CODE REVIEW

**Comment by jacoblee93** [comment]
Let me know when this is ready for review!

**Review by justinwang189 — COMMENTED** [review]
I pulled in this PR locally and found small issue in the constructor

**Review by ElectricCodeGuy — COMMENTED** [review]
Seems to work without any issue when updating. Currently testing my project. Thank you

**Comment by jacoblee93** [comment]
Happy to take this over if you're busy - thank you so much for opening this!

**Comment by zackproser** [comment]
Also, I'm not sure about the addition of `pnpm-lock.yaml` to `.gitignore` - that's probably better done in a completely separate PR if at all?

**Comment by jacoblee93** [comment]
Hey @zackproser - looks like we're nearing the finish line? Some high priority requests for Edge support and would absolutely love to get this out today, happy to take over.

File: langchain/src/vectorstores/pinecone.ts [hunk 1]
```diff
@@ -55,11 +62,18 @@ export class PineconeStore extends VectorStore {
...
+    })
+
     this.embeddings = embeddings;
-    const { namespace, pineconeIndex, textKey, filter, ...asyncCallerArgs } =
-      args;
+    const { namespace, pineconeIndex, pineconeIndexName, textKey, filter, ...asyncCallerArgs } = args;
     this.namespace = namespace;
-    this.pineconeIndex = pineconeIndex;
+    this.pineconeIndexName = args.pineconeIndexName ?? ''
+    this.pineconeIndex = pinecone.Index(this.pineconeIndexName)
```
[the remark this answers is not in the document]
Comment (zackproser) [reply langchain/src/vectorstores/pinecone.ts]: Thanks @justinwang189 - fixed! Not sure what I was thinking there.

**Comment by zackproser** [comment]
@jacoblee93 @justinwang189  Okay - the tests are now passing for me locally as is `yarn:lint` and I've updated the docs and `PineconeStore` according to review feedback. 

Please LMK how this looks to you when you have a moment and thanks for your help so far and patience! :bow:

**Comment by jacoblee93** [comment]
@zackproser looks good! Fixed up lint format and test idempotency, feel free to take a final look.

**Comment by zackproser** [comment]
> @zackproser looks good! Fixed up lint format and test idempotency, feel free to take a final look.

Thanks so much! LGTM :+1:

**Comment by jacoblee93** [comment]
Thanks so much!

**Comment by zackproser** [comment]
> Thanks so much!

Thank you so much for all your help! :raised_hands:

**Pull request merged**